### PR TITLE
first_branch

### PR DESCRIPTION
**Pull request by danikkrapivnitskiy**

### CODE REVIEW

File: JSTutorial/hw_6/task_2.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+4. У вас есть массив со скобками, предположим [ ‘(‘, ‘)’, ‘(‘, ‘)’, ‘)’], количество элементов и последовательность может быть разной.
+  Нужно выяснить, у каждой ли скобки есть соответствующая пара (открывающая и закрывающая).
+  Усложнение: в массиве могут быть вложены еще массивы на разной глубине. (ПОПРОБУЙТЕ МЕТОД .flat(), изучите как он работает с разными глубинами вложенности)
+  Вернуть нужно всё также есть ли у каждой скобочки соответствующая пара. 
+  Пример:
+  const arr = [[['(']], ')', '(', ')', ')', ['(', ['('], [')']]]
+
+ */
+//1
+  const sentenseStr = "I am the best AQA ever!";
```
Comment (VladimirLQA) [inline JSTutorial/hw_6/task_2.js:22]: Pay attention to task discription
`Преобразуйте строку таким образом, 
  чтобы вместо каждой буквы была цифра, показывающая сколько раз эта буква встречается в предложении.`

File: JSTutorial/hw_6/task_2.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  const alphabet = 'abcdefghijklmnopqrstuvwxyz';
+  const mapToNumber = [...sentenseStr].map(symbol => alphabet.includes(symbol.toLowerCase()) ? alphabet.indexOf(symbol.toLowerCase()) : symbol)
+  console.log(...mapToNumber);
```
Comment (VladimirLQA) [inline JSTutorial/hw_6/task_2.js]: We already know how to use functions, so it's time to use them :)

File: JSTutorial/hw_6/task_2.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+//2
+  const prices = [64, 7556, 345, 7556, 345, 7556, 345, 7556, 433, 345, 756, 123, 942, 3112, 421, 9341, 1212, 8, 43, 41, 345, 341, 21, 321, 123];
+  const avgPrice = prices.reduce((result, number) => result += number / prices.length, 0).toFixed(2);
```
Comment (VladimirLQA) [inline JSTutorial/hw_6/task_2.js]: The description does not mention float value as a result

File: JSTutorial/hw_6/task_2.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  console.log(avgPrice)
+
+//3
+  const words = ["sdfsdfsdfaaa", "lknjckbvhjibeiru", "oiusdfi"]
```
Comment (VladimirLQA) [inline JSTutorial/hw_6/task_2.js:33]: Use function syntax

File: JSTutorial/hw_6/task_2.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    result.set(word, count);
+    return result;
+  }, new Map());
+
+  const afterMap = [...filteredArr.entries()];
+  let max = 0;
+  let sortedArray = [];
+  for (let i = 0; i < afterMap.length; i++) {
+    if (afterMap[i][1] >= max) {
```
Comment (VladimirLQA) [inline JSTutorial/hw_6/task_2.js]: In this way you do not sort them, in the result I got same position for each word as it was
Try to run this case, for clarity
`const words = ['aai3', 'afd1', 'fsou2']`

File: JSTutorial/hw_6/task_2.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        sortedArray.unshift(afterMap[i][0]);
+    }
+  }
+
+  console.log(sortedArray);
+
+//4
+const arr = [[['(']], ')', '(', ')', ')', ['(', ['('], [')']]]
```
Comment (VladimirLQA) [inline JSTutorial/hw_6/task_2.js:41]: Use function syntax :)

**Review by VladimirLQA — CHANGES_REQUESTED** [review]

File: JSTutorial/hw_6/task_3.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  Вернуть массив с сугубо уникальными карточками. Реализовать методом SET.
+  Разобраться, как считать данные из файла КОДОМ, а не копировать ручками.
+ */
+
+//5
+let arr = [1, 2, 7, 5, 10, 5, 3, 4, 8, 9, 12]
+const missingNumber = [...new Set(arr)]
+    .sort((a, b) => a - b)
+    .filter(x => typeof x === "number" && x !==1 && !arr.includes(x - 1))
+    .map(x => x - 1);
```
Comment (anatoly-karpovich) [inline JSTutorial/hw_6/task_3.js]:
```suggestion
     .find((num, i, arr) => arr.indexOf(num) !== (num - 1)) - 1
```

**Review by anatoly-karpovich — APPROVED** [review]

File: JSTutorial/hw_6/task_2.js (hunk 6 above)
Comment (danikkrapivnitskiy) [reply JSTutorial/hw_6/task_2.js:41]: I did it, no?

arr.flat(Infinity).map(symbol => symbol === ")" ? count++ : count--);

File: JSTutorial/hw_6/task_2.js (hunk 5 above)
Comment (danikkrapivnitskiy) [reply JSTutorial/hw_6/task_2.js]: result is : [ 'fsou2', 'afd1', 'aai3' ]

**Pull request merged**